src/system.cr

```diff
@@ -22,4 +22,22 @@ module System
   def self.cpu_count
     Crystal::System.cpu_count
   end
+
+  # Returns the soft and hard file descriptor limits for the current process
+  #
+  # ```
+  # System.file_descriptor_limit # => { 1024, 524288 }
+  # ```
+  def self.file_descriptor_limit
+    Crystal::System.file_descriptor_limit
+  end
+
+  # Sets the soft file descriptor limits for the current process
+  #
+  # ```
+  # System.file_descriptor_limit = 4096
+  # ```
+  def self.file_descriptor_limit=(limit)
```

Comment:
If this is an int, can we add a type restriction?

Reply:
```suggestion
  def self.file_descriptor_limit=(limit : Int32)
```

Reply:
`UInt32` technically doesn't cut it the value is 64 bits wide on 64-bit platforms. In practice, this shouldn't be relevant, though. So we can perhaps stay with 32 bits. But maybe it should be signed instead? We try to avoid unsigned integer types in stdlib APIs.

Reply:
In any case, the same type should be applied as return type of the getter method.

Reply:
I was thinking just Int

Reply:
For the param type restriction, that would be good. But what about the return type?